### PR TITLE
Typescript Type Improvements

### PR DESCRIPTION
**Pull request by thejustinwalsh**
##### Description of change

Refactor and improve Typescript types

Fixes: 
- refactor PixiReactNode & PixiElement and fixes a bug with children
- draw prop is only available on Graphics
- fixes leaked types from ContructorOverrides
- adds some new utility types for readability
- adds tests for JSX

##### Pre-Merge Checklist

- [ x] Tests and/or benchmarks are included
- [ ] Documentation is changed or added
- [ x] Lint process passed (`npm run lint`)
- [ x] Tests passed (`npm run test`)

### CODE REVIEW

File: src/helpers/applyProps.ts [hunk 1]
```diff
@@ -85,7 +85,7 @@ export function applyProps(
         let currentInstance = instance;
         let targetProp = currentInstance[key];
 
-        if ((key === 'draw') && (typeof value === 'function'))
+        if ((key as string === 'draw') && (typeof value === 'function'))
```
Comment (thejustinwalsh) [inline src/helpers/applyProps.ts:88]: Not my favorite hack, after refactoring the draw callback prop on the types, but it does work. 🤣

**Comment by thejustinwalsh** [comment]
I think I broke the typescript type emitting when I attempted to fix the tests, My bad.
Working on an update now.

**Comment by thejustinwalsh** [comment]
Pulling `https://pkg.csb.dev/pixijs/pixi-react/commit/26b969aa/@pixi/react` into my local sandbox is working swimmingly!

For whatever reason, the CodeSanbox still tells me it can not find the types for the module!? 🤷

I'm pretty confident this is finally ready, but I could use your thoughts on the code sandbox issue.

**Review by trezy — APPROVED** [review]
LGTM!

**Pull request merged**